xc-admin/packages/crank-executor/src/index.ts

```diff
@@ -0,0 +1,77 @@
+import {
+  Commitment,
+  Connection,
+  Keypair,
+  PublicKey,
+  Transaction,
+} from "@solana/web3.js";
+import SquadsMesh, { DEFAULT_MULTISIG_PROGRAM_ID, getIxPDA } from "@sqds/mesh";
+import * as fs from "fs";
+import NodeWallet from "@project-serum/anchor/dist/cjs/nodewallet";
+import { getProposals } from "xc-admin-common";
+import BN from "bn.js";
+import { AnchorProvider } from "@project-serum/anchor";
+
+export function envOrErr(env: string): string {
+  const val = process.env[env];
+  if (!val) {
+    throw new Error(`environment variable "${env}" must be set`);
+  }
+  return String(process.env[env]);
+}
+
+const SOLANA_CLUSTER_URL: string = envOrErr("SOLANA_RPC_URL");
+const SOLANA_CONNECTION_COMMITMENT: Commitment =
+  (process.env.SOLANA_CONNECTION_COMMITMENT as Commitment) ?? "confirmed";
+const MULTISIG_VAULT: PublicKey = new PublicKey(envOrErr("SOLANA_RPC_URL"));
+const KEYPAIR: Keypair = Keypair.fromSecretKey(
+  Uint8Array.from(
+    JSON.parse(fs.readFileSync(envOrErr("KEYPAIR_PATH"), "ascii"))
+  )
+);
+
+async function run() {
+  const squad = new SquadsMesh({
+    connection: new Connection(
+      SOLANA_CLUSTER_URL,
+      SOLANA_CONNECTION_COMMITMENT
+    ),
+    wallet: new NodeWallet(KEYPAIR),
+    multisigProgramId: DEFAULT_MULTISIG_PROGRAM_ID,
+  });
+  const proposals = await getProposals(
+    squad,
+    MULTISIG_VAULT,
+    undefined,
+    "executeReady"
+  );
+
+  for (const proposal of proposals) {
+    // If we have previously cancelled because the proposal was failing, don't attempt
+    if (proposal.cancelled.length == 0) {
+      for (let i = proposal.executedIndex; i < proposal.instructionIndex; i++) {
+        const transaction = new Transaction().add(
+          await squad.buildExecuteInstruction(
+            proposal.publicKey,
+            getIxPDA(proposal.publicKey, new BN(i), squad.multisigProgramId)[0]
+          )
+        );
+
+        try {
+          await new AnchorProvider(squad.connection, squad.wallet, {
+            commitment: SOLANA_CONNECTION_COMMITMENT,
+            preflightCommitment: SOLANA_CONNECTION_COMMITMENT,
+          }).sendAndConfirm(transaction, [KEYPAIR]);
+        } catch (error) {
+          // Mark the transaction as cancelled if we failed to run it
+          await squad.cancelTransaction(proposal.publicKey);
```

Comment:
I think you need to handle the errors better here. For instance you should handle balance errors explicitly. There could be RPC errors and for that it's good to retry a couple of times before giving up on the transaction.

If checking these stuff is difficult I would suggest exponential backoff without cancelling. People will figure it out on the ui and will proceed to cancellation if necessary.

and... hmm... one question, why not letting people execute it on the ui? without us cranking this part? the last signer can also trigger the execute.